controllers/operatorconfig_controller.go

```diff
@@ -18,15 +18,39 @@ package controllers
 
 import (
 	"context"
+	"fmt"
+	"reflect"
+
 	"github.com/go-logr/logr"
+	cranev1alpha1 "github.com/konveyor/mtrho-operator/api/v1alpha1"
+	consolev1alpha1 "github.com/openshift/api/console/v1alpha1"
+	pipelinev1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+	"k8s.io/apimachinery/pkg/types"
 	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
 	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
 
-	cranev1alpha1 "github.com/konveyor/mtrho-operator/api/v1alpha1"
+const (
+	Finalizer       = "openshift.konveyor.crane"
+	OwnerLabel      = "crane.konveyor.io/ownerref"
+	OwnerConfigName = "openshift-migration"
```

Comment:
is this constant for all CRs?

How do we handle the effect of 2 OperatorConfig CRs creating two different sets of resources? or conversely 2 different OperatorConfig CRs reconciling on the same set of resources?

Reply:
Initially, we agreed upon only one CR would get created. So the code written will overlook and not reconcile if any other names CR is created since that wouldn't be desired behavior.

Reply:
fair. Just to be safe, I would recommend, creating the resource with specific owner name so that in the future if we have a need to create 2 CRs in the same namespace, we don't have to do a massive refactor then.

Reply:
Ack, this will need a little planning as well because I am not clear on what resources could be shared and what resources should be CR-dependent.  

For example, proxy and UI could be shared but cluster tasks should be owned by only one `operatorConfig` CR?

We could discuss this at length later.

Reply:
Would you be open to creating an issue so this does not die the pitiful death of a review comment :) 

I would recommend detailing the concerns about shared and owned resources there.

Reply:
You got it https://github.com/konveyor/mtrho-operator/issues/10